api_v2/stream.go

```diff
@@ -2,135 +2,187 @@
-    if err != nil {
-        return nil, err
-    }
+	streams, err := s.GetEnrolledOrPublicLiveStreams(a.db, &uID)
+	if err != nil {
+		return nil, err
+	}
 
-    resp := make([]*protobuf.Stream, len(streams))
-    for i, stream := range streams {
+	resp := make([]*protobuf.Stream, len(streams))
+	for i, stream := range streams {
 
-        c, err := s.GetCourseById(a.db, stream.CourseID)
-        if err != nil {
-            return nil, err
-        }
-                
-        if err := h.SignStream(stream, c, uID); err != nil {
-            return nil, err
-        }
+		c, err := s.GetCourseById(a.db, stream.CourseID)
+		if err != nil {
+			return nil, err
+		}
 
-        s, err := h.ParseStreamToProto(stream)
-        if err != nil {
-            return nil, err
-        } 
-        resp[i] = s
+		if err := h.SignStream(stream, c, uID); err != nil {
+			return nil, err
+		}
 
-    }
+		s, err := h.ParseStreamToProto(stream)
+		if err != nil {
+			return nil, err
+		}
+		resp[i] = s
 
-    return &protobuf.GetNowLiveResponse{Stream: resp}, nil
+	}
+
+	return &protobuf.GetNowLiveResponse{Stream: resp}, nil
 }
 
 func (a *API) GetThumbsVOD(ctx context.Context, req *protobuf.GetThumbsVODRequest) (*protobuf.GetThumbsVODResponse, error) {
-    a.log.Info("GetThumbsVOD")
-    
-    s, err := a.handleStreamRequest(ctx, req.StreamID)
-    if err != nil {
-        return nil, err
-    }
-
-    path, err := model.Stream.GetLGThumbnail(*s)
-    if err != nil {
-        path = "/thumb-fallback.png"
-    }
-
-    return &protobuf.GetThumbsVODResponse{Path: path}, nil
+	a.log.Info("GetThumbsVOD")
+
+	s, err := a.handleStreamRequest(ctx, req.StreamID)
+	if err != nil {
+		return nil, err
+	}
+
+	path, err := model.Stream.GetLGThumbnail(*s)
+	if err != nil {
+		path = "/thumb-fallback.png"
+	}
+
+	return &protobuf.GetThumbsVODResponse{Path: path}, nil
 }
 
 func (a *API) GetThumbsLive(ctx context.Context, req *protobuf.GetThumbsLiveRequest) (*protobuf.GetThumbsLiveResponse, error) {
-    a.log.Info("GetThumbsLive")
-    
-    s, err := a.handleStreamRequest(ctx, req.StreamID)
-    if err != nil {
-        return nil, err
-    }
-
-    path := pathprovider.LiveThumbnail(fmt.Sprintf("%d", s.ID))
-    if path == "" {
-        path = "/thumb-fallback.png"
-    }
-
-    
-    return &protobuf.GetThumbsLiveResponse{Path: path}, nil
-}
\ No newline at end of file
+	a.log.Info("GetThumbsLive")
+
+	s, err := a.handleStreamRequest(ctx, req.StreamID)
+	if err != nil {
+		return nil, err
+	}
+
+	path := pathprovider.LiveThumbnail(fmt.Sprintf("%d", s.ID))
+	if path == "" {
+		path = "/thumb-fallback.png"
+	}
+
+	return &protobuf.GetThumbsLiveResponse{Path: path}, nil
+}
+
+func (a *API) GetProgress(ctx context.Context, req *protobuf.GetProgressRequest) (*protobuf.GetProgressResponse, error) {
+	a.log.Info("GetStreamProgress")
+
+	uID, err := a.getCurrentID(ctx)
+	if err != nil && err.Error() != "missing cookie header" {
```

Comment:
`err != nil && err.Error() != "missing cookie header"` is used to allow  authenticated users (have valid jwt) as well as non-logged in users (missing jwt / cookie header) access to a certain method (e.g., GetPublicCourses or GetStream). In this case, GetProgress should only be accessed by authenticated users, so replace this with
```go
if err != nil {
  return nil, e.WithStatus(http.StatusUnauthorized, err)
}
```